activerecord/lib/active_record/connection_adapters/abstract/schema_definitions.rb

```diff
@@ -822,6 +832,21 @@ def check_constraint(*args)
       def remove_check_constraint(*args)
         @base.remove_check_constraint(name, *args)
       end
+
+      private
+        def raise_on_if_exist_options(options)
+          unrecognized_option, _value = options.find do |key, value|
+            (key == :if_exists || key == :if_not_exists) && !!value
+          end
```

Comment:
I had a hard time reading this, what about something like:

```ruby
unrecognized_option = options.keys.include?(:if_exists, :if_not_exists)
```

I'm not sure we should suggest a conditional to use, seems like overkill to me. Not sure what anyone else thinks.

Reply:
I'm torn on the suggestion because being helpful is helpful, but there are at least theoretical race-type considerations when compared to a "true" `IF EXISTS` clause on the statement itself. That level of detail doesn't belong here, but does making the suggestion without such a caveat imply they're more equivalent than they are?

Perhaps some form of "is not supported inside a change_table block" would be enough to at least hint them toward a "take it out of the block" solution, so we can avoid leaving people feeling stuck on how to progress even without explicitly recommending how they solve it?

Reply:
I'm overthinking it, but this was my thought process on putting the suggestion in the exception message:

1. Until I took a look at this problem, I wasn't aware that inside of a `change_table(:table) do |t|` block, `t.column_exists?` was even a valid method. If I received an exception saying that `t.string(:name, if_not_exist: true)` was invalid inside a change_table block, without the suggestion to move to an inline `if/unless` clause, I would have instinctively unrolled the change_table block into several separate `add_column(:table, ...)` expressions.
2. It's likely that this exception will be encountered both by people writing new migrations and by people/teams replaying migrations that have been committed in the past. When changing existing migrations, you'd want to have high confidence that you're not changing their fundamental behavior. Explicitly suggesting moving from a keyword option to an inline if/unless clause would keep the migration in the same "shape" (a change_table block with mostly the same method calls on `t` inside it), and the diff between versions would be easy to verify; whereas unrolling the block into several individual alteration expressions changes how the code reads and there's a lot of mental overhead to ensure that you haven't introduced an unwanted change.

Reply:
> It's likely that this exception will be encountered both by people writing new migrations and by people/teams replaying migrations that have been committed in the past

Ugh... that's a really good point. Technically we should push this through migration versioning, so it only applies to migrations generated with future Rails versions. 🤔

Reply:
Oh that sounds like a good idea, I will look up how to do that :)